src/ui/components/pages/MyServices/MyServices.tsx

```diff
@@ -244,21 +254,38 @@ export function MyServices(props: Props) {
         string | undefined
     >();
 
-    const [isDialogOpen, setIsDialogOpen] = useState(false);
+    const [isDialogDeleteOpen, setIsDialogDeleteOpen] = useState(false);
 
     const onRequestDelete = useConstCallback<MyServicesCardsProps["onRequestDelete"]>(
         ({ serviceId }) => {
             setServiceIdRequestedToBeDeleted(serviceId);
-            setIsDialogOpen(true);
+            setIsDialogDeleteOpen(true);
+        }
+    );
+    const onRequestUpgrade = useConstCallback<MyServicesCardsProps["onRequestUpgrade"]>(
+        async (params: UpgradeParams) => {
+            const formFieldsValueDifferentFromDefault = restorableContainerConfigs.find(
+                e => e.restorableLaunchPackageConfig.name === params.name
+            )?.restorableLaunchPackageConfig.formFieldsValueDifferentFromDefault;
+            assert(formFieldsValueDifferentFromDefault !== undefined);
+            await launcher.reset();
+            await launcher.initialize({
+                catalogId: params.catalogId!,
+                packageName: params.packageName!,
+                name: params.name,
+                formFieldsValueDifferentFromDefault
+            });
+
+            await launcher.launch();
+            runningService.update();
```

Comment:
This should be a thunk (core function). This is not the responsability of the UI to perform such complex logic.